packages/grid/src/vaadin-grid-column-mixin.js

```diff
@@ -0,0 +1,887 @@
+         *
+         * @private
+         */
+        __gridColumnElement: {
+          type: Boolean,
+          value: true,
+        },
+      };
+    }
+
+    static get observers() {
+      return [
+        '_widthChanged(width, _headerCell, _footerCell, _cells.*)',
+        '_frozenChanged(frozen, _headerCell, _footerCell, _cells.*)',
+        '_frozenToEndChanged(frozenToEnd, _headerCell, _footerCell, _cells.*)',
+        '_flexGrowChanged(flexGrow, _headerCell, _footerCell, _cells.*)',
+        '_textAlignChanged(textAlign, _cells.*, _headerCell, _footerCell)',
+        '_orderChanged(_order, _headerCell, _footerCell, _cells.*)',
+        '_lastFrozenChanged(_lastFrozen)',
+        '_firstFrozenToEndChanged(_firstFrozenToEnd)',
+        '_onRendererOrBindingChanged(_renderer, _cells, _bodyContentHidden, _cells.*, path)',
+        '_onHeaderRendererOrBindingChanged(_headerRenderer, _headerCell, path, header)',
+        '_onFooterRendererOrBindingChanged(_footerRenderer, _footerCell)',
+        '_resizableChanged(resizable, _headerCell)',
+        '_reorderStatusChanged(_reorderStatus, _headerCell, _footerCell, _cells.*)',
+        '_hiddenChanged(hidden, _headerCell, _footerCell, _cells.*)',
+        '_rowHeaderChanged(rowHeader, _cells.*)',
+      ];
+    }
+
+    /**
+     * @return {!Grid | undefined}
+     * @protected
+     */
+    get _grid() {
+      if (!this._gridValue) {
+        this._gridValue = this._findHostGrid();
+      }
+      return this._gridValue;
+    }
+
+    /**
+     * @return {!Array<!HTMLElement>}
+     * @protected
+     */
+    get _allCells() {
+      return []
+        .concat(this._cells || [])
+        .concat(this._emptyCells || [])
+        .concat(this._headerCell)
+        .concat(this._footerCell)
+        .filter((cell) => cell);
+    }
+
+    /** @protected */
+    connectedCallback() {
+      super.connectedCallback();
+
+      // Adds the column cells to the grid after the column is attached
+      requestAnimationFrame(() => {
+        // Skip if the column has been detached
+        if (!this._grid) {
+          return;
+        }
+
+        this._allCells.forEach((cell) => {
+          if (!cell._content.parentNode) {
+            this._grid.appendChild(cell._content);
+          }
+        });
+      });
+    }
+
+    /** @protected */
+    disconnectedCallback() {
+      super.disconnectedCallback();
+
+      // Removes the column cells from the grid after the column is detached
+      requestAnimationFrame(() => {
+        // Skip if the column has been attached again
+        if (this._grid) {
+          return;
+        }
+
+        this._allCells.forEach((cell) => {
+          if (cell._content.parentNode) {
+            cell._content.parentNode.removeChild(cell._content);
+          }
+        });
+      });
+
+      this._gridValue = undefined;
+    }
+
+    /** @protected */
+    ready() {
+      super.ready();
+
+      processTemplates(this);
```

Comment:
Not sure if we want to also have this in the Lit based version. Not a blocker, can be preserved for now.

Reply:
Good point. The old `<template>` API still works with the Lit-based grid using the adapter though (it shows you `WARNING: <template> inside <vaadin-grid-column> is deprecated. Use a renderer function instead`). Let's remember to remove all the legacy adapters in the next major.